### PR TITLE
chore: add API deprecations mechanism

### PR DESCRIPTION
**Pull request by AlekSi**
Refs #4576.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4611)

### CODE REVIEW

File: api/common/common.proto [hunk 1]
```diff
@@ -5,8 +5,44 @@ package common;
...
 import "google/rpc/status.proto";
 
+// An alternative to using options could be extracting versions from comments.
+// Unfortunately, they are not available: https://github.com/golang/protobuf/issues/1134
+// Also, while option numbers can be the same,
+// names should be different: https://github.com/protocolbuffers/protobuf/issues/4861
+
+extend google.protobuf.MessageOptions {
+  // Indicates the Talos version when this deprecated message will be removed from API.
+  string remove_deprecated_message = 93117;
```
Comment (AlekSi) [inline api/common/common.proto:18]: Explaining the significance of that number is left as an exercise to the reader 😆

**Review by Unix4ever — APPROVED** [review]
nice!

**Pull request merged**